### PR TITLE
Add createdBy field to TransactionItem, TransactionEvent

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds `createdBy` field to `TransactionItem` and `TransactionEvent`. As described in 
#10350 section 2.2.2 and 2.2.3

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by kadewu — APPROVED** [review]

File: saleor/graphql/payment/types.py [hunk 1]
```diff
@@ -323,6 +325,14 @@ def resolve_external_url(root: models.TransactionEvent, info):
     def resolve_name(root: models.TransactionEvent, info):
         return root.message
 
+    @staticmethod
+    def resolve_created_by(root: models.TransactionItem, info):
```
Comment (fowczarek) [inline saleor/graphql/payment/types.py:329]: Shouldn't be wrapped in any permissions?

File: saleor/graphql/payment/types.py [hunk 2]
```diff
@@ -434,26 +426,11 @@ def resolve_events(root: models.TransactionItem, info):
...
-            return None
-
-        if not root.user_id:
-            return None
-
-        return UserByUserIdLoader(info.context).load(root.user_id).then(_resolve_user)
-
-    @staticmethod
-    def resolve_app(root: models.TransactionItem, info):
+    def resolve_created_by(root: models.TransactionItem, info):
```
Comment (fowczarek) [inline saleor/graphql/payment/types.py:429]: Shouldn't be wrapped in any permissions?

File: saleor/graphql/payment/types.py (hunk 1 above)
Comment (korycins) [reply saleor/graphql/payment/types.py:329]: I've already added separate to-do step to define a proper permission for this field as I need to re-think what needs to be required here. For now, let's leave it as it is but thanks for catching this.
Ref: https://github.com/saleor/saleor/pull/10350
image

File: saleor/graphql/payment/types.py (hunk 2 above)
Comment (korycins) [reply saleor/graphql/payment/types.py:429]: https://github.com/saleor/saleor/pull/11534#discussion_r1054132965

**Pull request merged**